smithy-protocols-traits/src/main/java/software/amazon/smithy/protocols/traits/Rpcv2Trait.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.protocols.traits;
+ 
+import java.util.ArrayList;
+import java.util.List;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.ObjectNode;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.AbstractTrait;
+import software.amazon.smithy.model.traits.AbstractTraitBuilder;
+import software.amazon.smithy.utils.ListUtils;
+
+public final class Rpcv2Trait extends AbstractTrait {
+
+    private static final String HTTP = "http";
+    private static final String EVENT_STREAM_HTTP = "eventStreamHttp";
+    private static final String FORMAT = "format";
+
+    private final List<String> http;
+    private final List<String> eventStreamHttp;
+    private final List<String> format;
+
+    public static final ShapeId ID = ShapeId.from("smithy.protocols#smithyRpcV2");
+
+    // package-private constructor (at least for now)
```

Comment:
This comment is currently wrong, but I think it can go back to being package private.